File: LeetCode/Medium/1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0;
    public int goodNodes(TreeNode root) {
        
        this.ans = 0;

        this.dfs(root, Integer.MIN_VALUE);


        return this.ans;
    }

    private void dfs(TreeNode root, int curMax) {
        if(root == null) return;

        curMax = Math.max(curMax, root.val);

        if(curMax <= root.val) this.ans++;

        this.dfs(root.left, curMax);
        this.dfs(root.right, curMax);
    }
}

File: LeetCode/Medium/1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    var ans = 0
    fun goodNodes(root: TreeNode?): Int {
        
        if(root == null) return 0;
        
        this.ans = 0
        
        this.dfs(root, Integer.MIN_VALUE)
        
        return this.ans
    }
    
    private fun dfs(root: TreeNode?, receiveMax: Int) {
        if(root == null) return
        
        val curMax = Math.max(receiveMax, root.`val`)

        if(curMax <= root.`val`) this.ans++

        this.dfs(root.left, curMax)
        this.dfs(root.right, curMax)
    }

}